projects/reservoir/fv3/era_process.py

```diff
@@ -0,0 +1,177 @@
+            make_hgrid \
+            --grid_type regular_lonlat_grid \
+            --grid_name /work/era5_lonlat_grid \
+            --nxbnds 2 \
+            --nybnds 2 \
+            --xbnds -180,180 \
+            --ybnds -90,90 \
+            --nlon 720\
+            --nlat 362")
+        
+    os.system("sudo docker run \
+        -v $(pwd)/fregrid-example:/work \
+        us.gcr.io/vcm-ml/post_process_run:latest \
+        make_solo_mosaic \
+        --num_tiles 1\
+        --dir /work \
+        --tile_file era5_lonlat_grid.nc \
+        --mosaic_name /work/era5_lonlat_grid_mosaic")
+                
+def check_time_steps_complete(path, var):    
+    #check if data is complete (all time steps are there)
+    time_step_list = []
+    for f in glob.glob(path+ var +'/downloaded/*.nc'):
+        dataset = xr.open_dataset(f)
+        time_step_list.append(dataset['time'].data)
+    dataset.close()
+    arrs = list(np.concatenate(time_step_list))
+    arrs.sort()
+    for i,date in enumerate(arrs[1:]):
+        if not (arrs[i]+np.timedelta64(1,'D')==date):
+            print('missing day after:', arrs[i], i)
+    print('all time steps checked.')
+    
+def merge_into_file(path,var,i,day):
+    #create one file per weekday
+    day_sublist = []
+    for f in glob.glob(path+ var +'/downloaded/*.nc'):
+        dataset = xr.open_dataset(f)
+        for time in range(len(dataset.time)):
+            data_at_day = dataset.isel(time=slice(time,time+1))
+            if data_at_day.time.dt.weekday[0] == i:
+                day_sublist.append(data_at_day)
+
+        dataset.close()
+    merged_data = xr.concat(day_sublist,dim='time')
+    sorted_data = merged_data.sortby('time')
+    sorted_data.to_netcdf(path + var + '/merged/' + var + '_' + day +'.nc')
+    save_nc_int32_time(path + var + '/merged/'+var+'_'+day+'.nc', '/home/paulah/fregrid-example/'+var+'_'+day+'_i32_time.nc')
+    #os.system("cp "+path+var+"/merged/"+var+"_"+day+"_i32_time.nc /home/paulah/fregrid-example")
+    
+def save_nc_int32_time(infile, outfile):
+    in_nc = nc.Dataset(infile, "r")
+    
+    # load original time variable
+    in_time = in_nc.variables["time"]
+    in_time_values = in_time[:]
+    as_dt = cftime.num2date(in_time_values, in_time.units, calendar=in_time.calendar)
+    as_julian = cftime.date2num(as_dt, in_time.units, calendar="julian")
+    in_nc.close()
+    
+    # save new file without time coordinate
+    in_ds = xr.open_dataset(infile)
+    in_ds.drop("time").to_netcdf(outfile)
+    
+    # add adjusted time dimension to the new file
+    
+    out_nc = nc.Dataset(outfile, "a")
+    try:
+        out_time = out_nc.createVariable("time", np.int32, ("time",))
+        out_time[:] = as_julian
+        for attr in in_time.ncattrs():
+            if attr == "calendar":
+                value = "julian"
+            else:
+                value = in_time.getncattr(attr)
+            out_time.setncattr(attr, in_time.getncattr(attr))
+    except:
+        pass
```

Comment:
I agree there should be a print message here describing the error at the very least.  

The block is set mainly to be a catch all that encapsulates the operations on the netCDF, since any failure will corrupt the netCDF if the file handle is left open.  I didn't see a way to open the netCDF file with a context manager, which probably would be the proper way to handle this.  (`with nc.<open_file_context_function>(path): do operations`)  I didn't look very hard to see if anything like that existed.

Reply:
@frodre A question about your `save_nc_int32_time`: What is the purpose of `value`? Because I don't see it being used after being set.

Reply:
@paulaharder Yes, that's a bug.  `value` should be used in place of `in_time.getncattr(attr)` inside `setncattr`.